src/extras/gizmo/gizmo.js

```diff
@@ -258,9 +275,22 @@ class Gizmo extends EventHandler {
         this._device.canvas.addEventListener('pointermove', this._onPointerMove, true);
         this._device.canvas.addEventListener('pointerup', this._onPointerUp);
 
-        app.on('update', () => this._updateScale());
+        this._app.on('update', () => {
+            this._updatePosition();
+            this._updateRotation();
+            this._updateScale();
```

Comment:
Could we only update when position actually changed (simple distance-squared check against epsilon should suffice to not spam with update events)?